### PR TITLE
Update

### PR DESCRIPTION
**Pull request by olaferlandsen**
Features:
* Adding types for Typescript(compatible with Ionic2)
* Recompiling `ftp4j` and solve some problems in cordova
* Adding `Ftp` class writing in Typescript for Ionic2.
* Update README.md with some examples

Please add thanks to me 😄  bye!

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,9 +29,41 @@ Dependency:
...
+#### Requirements:
+
+You can access this plugin by typescript class `Ftp`.
+#### Example
+```typescript
+contructor (private ftp: Ftp) {
+    this.ftp.connect('hostname.com', 'username', 'password').then(() => {
+        this.ftp.upload('file.zip', '/var/www/file.zip').then(() => {
+            this.ftp.download('file.zip', '/var/www/file.zip').then(() => {
+                this.ftp.remove('/var/www/file.zip').then(() => {
```
Comment (xfally) [inline README.md]: remove => rm. keep name consistency please.

File: types/Ftp.ts [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+     *      this.ftp.isAvailible().then(() => {
+     *          this.ftp.connect('localhost', 'username', password'').then(() => {
+     *              this.ftp.remove('/var/www/', true).then(() => {
+     *                  // success
+     *              })
+     *          })
+     *      })
+     *  }
+     * */
+    public static remove (remote:string, recursive?:boolean):Promise<any>
```
Comment (xfally) [inline types/Ftp.ts]: suggest to keep name consistency. remove => rm, list => ls...

**Review by xfally — COMMENTED** [review]
I notice that you updated ftp4j-1.7.2.jar which uses LGPL. Could you introduce what have been modified?

**Comment by olaferlandsen** [comment]
I recompiled ftp4j because you compilation using a old compiling version. It causing some warnings when you build a apk file.

**Comment by olaferlandsen** [comment]
Hi @xfally I updated typescript version with rm, rmdir, ls, mkdir, and disconnect.
Maybe tomorrow I can write a functional full "example".
I think in new features like:
* `boolean` exists (`string` path) - Check if file or directory exists
* `boolean` isFile( `string` path ) - Check if is File
* `boolean` isDir( `string` path ) - Check if is Directory
* `float` getSize( `string` path ) - Get file size
* `string` pwd() - Get working directory
* `boolean` chmod( `string` path ) - Change file permissions
* `boolean` rename (`string` path) - Rename file or directory
* `integer` getPermisions( `string` path )

Whats do you think about it?

**Comment by xfally** [comment]
@olaferlandsen Thanks for the work! Could you `rebase` all commits into one commit (topic for typescript packaging) and then send the PR? so the git history will be clearer. If you don't know how to do, I can help to finish it via github `Squash and merge` option, but may make your branch unhappy.

Surely full example for typescript is welcome, just like the existing [demo.js](https://github.com/xfally/cordova-plugin-ftp/blob/master/demo.js).

New features? I'm sorry no plan for that. Why? Because the existing cmd `ls` can finish most features (`exists, isFile, isDir, getSize`) you want. Need `pwd, rename`? Maybe we can check (or modify) GoldRaccoon and ftp4j to support them, but not all ftp server support those features, so they are unconsidered in this ftp client. The remaining `chmod, getPermissions` seem not be defined in [rfc959](https://www.w3.org/Protocols/rfc959/) although some PC ftp tools have implemented them.

**Comment by olaferlandsen** [comment]
I'm so sorry, I don't know how I can `rebase` 😅

**Comment by xfally** [comment]
As said in issue #13, thanks for your work.

**Pull request closed without merging**